test/unit/flash_attention/CMakeLists.txt

```diff
@@ -26,52 +26,22 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-if(SYCL_INTEL_TARGET)
-
 set(CUTLASS_APPLICATIONS_DIR ${CMAKE_SOURCE_DIR}/applications)
 
-cutlass_test_unit_add_executable(
-  cutlass_test_unit_flash_attention_xe
-  xe_flash_attention_bf16.cpp
-  xe_flash_attention_fp16.cpp
-)
+add_subdirectory(flash_attention_prefill)
+add_subdirectory(flash_attention_decode)
 
-cutlass_test_unit_add_executable(
-  cutlass_test_unit_flash_decode_xe
-  xe_flash_decode_bf16.cpp
-  xe_flash_decode_fp16.cpp
-)
 
 add_custom_target(
   cutlass_test_unit_flash_attention
   DEPENDS
-  cutlass_test_unit_flash_attention_xe
-)
-
-add_custom_target(
-  cutlass_test_unit_flash_decode
-  DEPENDS
-  cutlass_test_unit_flash_decode_xe
+  cutlass_test_unit_flash_attention_prefill
```

Comment:
why do we still need the `cutlass_test_unit_flash_attention` target defined if `not SYCL_INTEL_TARGET`?

Maybe add a comment

Reply:
I have limited the flash_attention tests to be added only when `SYCL_INTEL_TARGET` is set.